src/scene/shader-lib/programs/lit-shader.js

```diff
@@ -1303,10 +1305,10 @@ class LitShader {
 
                     // area light
                     if (options.useClearCoat) {
-                        backend.append(`    ccSpecularLight += ccLTCSpecFres * get${shapeString}LightSpecular(litShaderArgs.clearcoat.worldNormal, dViewDirW) * dAtten * light${i}_color` + (usesCookieNow ? " * dAtten3" : "") + ";");
+                        backend.append(`    ccSpecularLight += ccLTCSpecFres * get${shapeString}LightSpecular(litShaderArgs.clearcoat.worldNormal, dViewDirW) * dAtten * light${i}_color * light${i}_specularColorFactor` + (usesCookieNow ? " * dAtten3" : "") + ";");
```

Comment:
Like Martin said, I would probably just change this chunk to do something like `... light${i}_color * ${specularSwitch}...` and then just set specularSwitch to 1 normally or 0 if we disable specular highlights.